core/node/prover_api/src/processor.rs

```diff
@@ -0,0 +1,196 @@
+        let mut conn = self.pool.connection().await.map_err(ProcessorError::Dal)?;
+
+        let previous_batch_metadata = conn
+            .blocks_dal()
+            .get_l1_batch_metadata(L1BatchNumber(l1_batch_number.checked_sub(1).unwrap()))
+            .await
+            .map_err(ProcessorError::Dal)?
+            .expect("No metadata for previous batch");
+
+        let header = conn
+            .blocks_dal()
+            .get_l1_batch_header(l1_batch_number)
+            .await
+            .map_err(ProcessorError::Dal)?
+            .unwrap_or_else(|| panic!("Missing header for {}", l1_batch_number));
+
+        let minor_version = header.protocol_version.unwrap();
+        let protocol_version = conn
+            .protocol_versions_dal()
+            .get_protocol_version_with_latest_patch(minor_version)
+            .await
+            .map_err(ProcessorError::Dal)?
+            .unwrap_or_else(|| {
+                panic!("Missing l1 verifier info for protocol version {minor_version}")
+            });
+
+        let batch_header = conn
+            .blocks_dal()
+            .get_l1_batch_header(l1_batch_number)
+            .await
+            .map_err(ProcessorError::Dal)?
+            .unwrap_or_else(|| panic!("Missing header for {}", l1_batch_number));
+
+        let eip_4844_blobs = match self.commitment_mode {
+            L1BatchCommitmentMode::Validium => Eip4844Blobs::empty(),
+            L1BatchCommitmentMode::Rollup => {
+                let blobs = batch_header.pubdata_input.as_deref().unwrap_or_else(|| {
+                    panic!(
+                        "expected pubdata, but it is not available for batch {l1_batch_number:?}"
+                    )
+                });
+                Eip4844Blobs::decode(blobs).expect("failed to decode EIP-4844 blobs")
+            }
+        };
+
+        let blob = WitnessInputData {
+            vm_run_data,
+            merkle_paths,
+            eip_4844_blobs,
+            previous_batch_metadata: L1BatchMetadataHashes {
+                root_hash: previous_batch_metadata.metadata.root_hash,
+                meta_hash: previous_batch_metadata.metadata.meta_parameters_hash,
+                aux_hash: previous_batch_metadata.metadata.aux_data_hash,
+            },
+        };
+
+        Ok(ProofGenerationData {
+            l1_batch_number,
+            witness_input_data: blob,
+            protocol_version: protocol_version.version,
+            l1_verifier_config: protocol_version.l1_verifier_config,
+        })
+    }
+
+    pub(crate) async fn verify_proof(
+        &self,
+        Path(l1_batch_number): Path<u32>,
+        Json(payload): Json<VerifyProofRequest>,
+    ) -> Result<(), ProcessorError> {
+        tracing::info!(
+            "Received request to verify proof for batch: {}",
+            l1_batch_number
+        );
+
+        let serialized_proof = bincode::serialize(&payload.0)?;
+        let expected_proof = bincode::serialize(
```

Comment:
When we fetch proof generation input, right now we pick the job that _doesn't_ have a proof, while here we expect the proof to exist. First of all, we should probably alter the logic for fetching the job (see also other comments on this topic), but secondly -- would it make sense to check if the batch is already proven in the DB before going to GCS?